internal/build/build.go

```diff
@@ -67,7 +77,15 @@ func init() {
 // CheckForUpdate checks if there is a newer version of the CLI than
 // the version currently being used. If so, it logs this information
 // to the console.
-func CheckForUpdate(ctx context.Context, logger logging.Logger, localizer localize.Localizer) {
+func CheckForUpdate(ctx context.Context, version string, logger logging.Logger, localizer localize.Localizer) {
+	if BuildSource != string(githubBuildSource) {
+		return
+	}
+	// prefix version with a v to correspond with Git tag
+	if !strings.HasPrefix(version, "v") {
+		version = "v" + version
+	}
```

Comment:
Turns out version checks had not been working because of differences in Git tag and version.

Reply:
Any way I can verify this locally? 

It looks like good fix

Reply:
```shell
BUILD_SOURCE=github RHOAS_VERSION=0.30.0 make binary
rhoas version
```